### PR TITLE
Sanitycheck: Support filtering based on DTS configuration data

### PR DESCRIPTION
**Pull request by nashif**
Cleanup some hardcoded samples and depend on DTS data instead making the sample
more generic.

### CODE REVIEW

**Review by tyrantlizard77 — APPROVED** [review]
LGTM, would like @galak blessing

**Review by spoorthik — APPROVED** [review]

**Comment by galak** [comment]
I want to think about this a little before approving/merging, not sure what the implications are we remove dt fixup, move to codegen, etc.

**Comment by nashif** [comment]
> I want to think about this a little before approving/merging, not sure what the implications are we remove dt fixup, move to codegen, etc.

we can always change to whatever method later and when such changes come, right now we are just checking for variables that are being used in samples anyways and those would need to be changed in any case depending on how we move forward. 
Right now the hardcoding in samples and coverage of tests is very limited because we have no other way of filtering.

File: samples/basic/disco/README.rst [hunk 1]
```diff
@@ -13,8 +13,12 @@ GPIO outputs of the MCU/board.
 Wiring
 ******
 
-The code may need some work before running on another board: set PORT,
-LED1 and LED2 according to the board's GPIO configuration.
+This sample should work on board with multiple built-in LEDs without any
```
Comment (dbkinder) [inline samples/basic/disco/README.rst]: This is contradictory, so let's simplify and say:
```
This sample may need some code changes to run on specific
boards by setting the PORT, LED0, and LED1 according to
the board's GPIO configuration.  As examples, see the instructions
below for some of our supported boards.
```

**Review by dbkinder — DISMISSED** [review]

**Review by erwango — APPROVED** [review]

**Review by galak — DISMISSED** [review]
Is there some way we can convey these filters are coming from DT data?  Would be helpful when we replace this with eDTS info that we know that.  Not sure if adding a 'dt_filter' in addition to 'filter'.

**Comment by nashif** [comment]
> Is there some way we can convey these filters are coming from DT data? Would be helpful when we replace this with eDTS info that we know that. Not sure if adding a 'dt_filter' in addition to 'filter'.

no, there is no way I can think of, the filter also supports environment variables, some variables from cmake, Kconfig etc. So DT is not special here.

We should not block this because of EDTS, unless we have a clear idea and a plan about when EDTS is landing.

File: samples/basic/blinky/sample.yaml [hunk 2]
```diff
@@ -3,4 +3,5 @@ sample:
 tests:
   test:
     tags: LED gpio
+    filter: LED0_GPIO_CONTROLLER
```
Comment (galak) [inline samples/basic/blinky/sample.yaml]: How about a comment like:
`# LED0_GPIO_CONTROLLER generated from dts`

Comment (galak) [inline samples/basic/blinky/sample.yaml]: Missing comment here.

File: samples/basic/rgb_led/sample.yaml [hunk 3]
```diff
@@ -2,7 +2,6 @@ sample:
   name: RGB LED
 tests:
   test:
-    # FIXME: We should remove those and just rely on depends_on
-    platform_whitelist: arduino_101 hexiwear_k64
+    filter: RED_PWM_LED_PWM_CONTROLLER
```
Comment (galak) [inline samples/basic/rgb_led/sample.yaml]: missing comment here.

**Review by galak — APPROVED** [review]

**Pull request merged**